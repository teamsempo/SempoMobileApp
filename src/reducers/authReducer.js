export const LOGIN_REQUEST = 'LOGIN_REQUEST';
export const LOGIN_PARTIAL = 'LOGIN_PARTIAL';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAILURE = 'LOGIN_FAILURE';
export const LOGOUT = 'LOGOUT';

export const REAUTH_REQUEST = 'REAUTH_REQUEST';

export const SET_PIN_REQUEST = 'SET_PIN_REQUEST';
export const SET_PIN_SUCCESS = 'SET_PIN_SUCCESS';
export const SET_PIN_FAILURE = 'SET_PIN_FAILURE';

export const VERIFY_PIN_REQUEST = 'VERIFY_PIN_REQUEST';
export const VERIFY_PIN_SUCCESS = 'VERIFY_PIN_SUCCESS';
export const VERIFY_PIN_FAILURE = 'VERIFY_PIN_FAILURE';

export const OPEN_APP_REQUEST = 'OPEN_APP_REQUEST';
export const OPEN_APP_SUCCESS = 'OPEN_APP_SUCCESS';
export const OPEN_APP_FAILURE = 'OPEN_APP_FAILURE';
export const BACKGROUND_APP = 'BACKGROUND_APP';
export const UPDATE_REQUIRE_PIN = 'UPDATE_REQUIRE_PIN';

//Partial Login Enumerable.
// TODO: Consider unifying with api response by either making API response an enumerable, or making this a set of bools
export const NEED_SET_PIN = 'NEED_SET_PIN';
export const NEED_OTP_VERIFY = 'NEED_OTP_VERIFY';
export const NEED_LOGIN_WITH_PIN = 'NEED_LOGIN_WITH_PIN';


const initialLoginState = {
    isRequesting: false,
    token: null,
    userId: null,
    firstName: null,
    lastName: null,
    transferAccountId: null,
    isVendor: null,
    isSupervendor: null,
    requestFeedbackQuestions: [],
    defaultFeedbackQuestions: [],
    deploymentName: null,
    termsAccepted: null,
    kycActive: null,
    denominations: {},

    secret: null,
    ecdsaPublic: null,
    localServerTimeDelta: null,
    pusherKey: null,
    displayDecimals: null,
    currencySymbol: null,
    currencyConversionRate: 1,
    transferUsages: null,
    NFCPositions: null,
    error: null,
    forgivingDeduct: false,
    supportSigValidation: false,
    partialState: null
};

export const login = (state = initialLoginState, action) => {
    switch (action.type) {
        case LOGIN_REQUEST:
            return {...state, isRequesting: true};
        case LOGIN_SUCCESS:
            return {...state,
                isRequesting: false,
                token: action.token,
                userId: action.userId,
                firstName: action.firstName,
                lastName: action.lastName,
                transferAccountId: action.transferAccountId,
                isVendor: action.isVendor,
                isSupervendor: action.isSupervendor,
                secret: action.secret,
                ecdsaPublic: action.ecdsaPublic,
                localServerTimeDelta: action.localServerTimeDelta,
                pusherKey: action.pusherKey,
                displayDecimals: action.displayDecimals,
                currencySymbol: action.currencySymbol,
                currencyConversionRate: action.currencyConversionRate,
                requestFeedbackQuestions: action.requestFeedbackQuestions,
                defaultFeedbackQuestions: action.defaultFeedbackQuestions,
                deploymentName: action.deploymentName,
                termsAccepted: action.termsAccepted,
                kycActive: action.kycActive,
                denominations: action.denominations,
                transferUsages: action.transferUsages,
                NFCPositions: action.NFCPositions,
                forgivingDeduct: action.forgivingDeduct,
                supportSigValidation: action.supportSigValidation
            };
        case LOGIN_PARTIAL:
            return {
                ...state,
                isRequesting: false,
                token: null,
                userId: null,
                partialState: action.partialState,
                error: action.error
            };
        case LOGIN_FAILURE:

            try {
                //We've found one of our backend's expected messages

                //This bit here is a hack because incorrect passwords get passed back from the delegator as 200 status codes
                if (typeof action.error === 'string') {
                    var error_message = action.error
                } else {
                    error_message = action.error.response.data.message
                }
            } catch (e) {
                error_message = 'unknown error'
            }

            console.log(error_message)

            return {...state, isRequesting: false, token: null, userId: null, error: error_message};
        case LOGOUT:
            return initialLoginState;
        default:
            return state;
    }
};



export const initialSetPinState = {
    isRequesting: false,
    success: false,
    error: null
};

export const setPin = (state = initialSetPinState, action) => {
    switch (action.type) {
        case SET_PIN_REQUEST:
            return {...state, isRequesting: true, error: null, success: false};
        case SET_PIN_SUCCESS:
            return {...state, isRequesting: false, success: true};
        case SET_PIN_FAILURE:

            try {
                var error_message = action.error.response.data.message
            } catch (e) {
                error_message = 'unknown error'
            }

            return {...state, isRequesting: false, error: error_message};
        default:
            return state;
    }
};

const initialVerifyPinState = {
    isRequesting: false,
    success: false,
    error: null,
};

export const verifyPin = (state = initialVerifyPinState, action) => {
    switch (action.type) {
        case OPEN_APP_REQUEST:
            //When the app is opened, reset pin verification state
            return initialVerifyPinState;
        case VERIFY_PIN_REQUEST:
            return {...state, isRequesting: true, error: null, success: false};
        case VERIFY_PIN_SUCCESS:
            return {...state, isRequesting: false, success: true};
        case VERIFY_PIN_FAILURE:
            return {...state, isRequesting: false, success: false, error: action.error};
        default:
            return state;
    }
};

export const initialAppState = {
    isRequesting: false,
    success: false,
    error: null,
    requirePIN: true,
    backgrounded: true,
};

export const openApp = (state = initialAppState, action) => {
    //TODO: consider renaming this reducer (and saga etc) to APP_STATE_CHANGE or something given we use it on close too
    switch (action.type) {
        case OPEN_APP_REQUEST:
            return {...state, isRequesting: true, success: false};
        case OPEN_APP_SUCCESS:
            return {...state, isRequesting: false, success: true};
        case OPEN_APP_FAILURE:
            return {...state, isRequesting: false, success: false, error: action.error};
        case BACKGROUND_APP:
            return {...state, isRequesting: false, backgrounded: action.backgroundedTime};
        case UPDATE_REQUIRE_PIN:
            return {...state, isRequesting: false, requirePIN: action.requirePIN};
        default:
            return state;
    }
};

// ACTIONS
export const loginRequest = (payload) => (
    {
        type: LOGIN_REQUEST,
        payload,
    }
);

export const logout = () => (
    {
        type: LOGOUT
    }
);

export const setPinRequest = (payload) => (
    {
        type: SET_PIN_REQUEST,
        payload
    }
);

export const verifyPinRequest = (payload) => (
    {
        type: VERIFY_PIN_REQUEST,
        payload
    }
);

export const openAppRequest = () => (
    {
        type: OPEN_APP_REQUEST,
    }
);

export const backgroundApp = (backgrounded) => (
    {
        type: OPEN_APP_REQUEST,
        backgrounded: backgrounded
    }
);

export const updateRequirePIN = (requirePIN) => (
    {
        type: UPDATE_REQUIRE_PIN,
        requirePIN
    }
);