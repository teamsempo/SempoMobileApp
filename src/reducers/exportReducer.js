export const RESET_EXPORT = 'RESET_EXPORT';
export const NEW_EXPORT_REQUEST = 'NEW_EXPORT_REQUEST';
export const NEW_EXPORT_SUCCESS = 'NEW_EXPORT_SUCCESS';
export const NEW_EXPORT_FAILURE = 'NEW_EXPORT_FAILURE';

export const initialExportState = {
    isRequesting: false,
    success: false,
    error: null,
    file_url: null,
};

export const newExportRequest = (state = initialExportState, action) => {
    switch (action.type) {
        case RESET_EXPORT:
            return initialExportState;
        case NEW_EXPORT_REQUEST:
            return {...state, isRequesting: true, error: null, success: false};
        case NEW_EXPORT_SUCCESS:
            return {...state, isRequesting: false, success: true, file_url: action.file_url};
        case NEW_EXPORT_FAILURE:

            try {
                var error_message = action.error.response.data.message
            } catch (e) {
                error_message = 'unknown error'
            }

            return {...state, isRequesting: false, error: error_message};
        default:
            return state;
    }
};


// Actions

export const newExport = ({date_range, email}) => (
    {
        type: NEW_EXPORT_REQUEST,
        date_range,
        email
    }
);
