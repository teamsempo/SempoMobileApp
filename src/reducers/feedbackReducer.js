export const RESET_NEW_FEEDBACK_DATA = 'RESET_NEW_FEEDBACK_DATA';
export const NEW_FEEDBACK_REQUEST = 'NEW_FEEDBACK_REQUEST';
export const NEW_FEEDBACK_SUCCESS = 'NEW_FEEDBACK_SUCCESS';
export const NEW_FEEDBACK_FAILURE = 'NEW_FEEDBACK_FAILURE';

export const initialNewFeedbackState = {
    isRequesting: false,
    success: false,
    error: null
};

export const newFeedbackData = (state = initialNewFeedbackState, action) => {
    switch (action.type) {
        case RESET_NEW_FEEDBACK_DATA:
            return initialNewFeedbackState;
        case NEW_FEEDBACK_REQUEST:
            return {...state, isRequesting: true, error: null, success: false};
        case NEW_FEEDBACK_SUCCESS:
            return {...state, isRequesting: false, success: true};
        case NEW_FEEDBACK_FAILURE:

            try {
                var error_message = action.error.response.data.message
            } catch (e) {
                error_message = 'unknown error'
            }

            return {...state, isRequesting: false, error: error_message};
        default:
            return state;
    }
};


// Actions

export const newFeedback = ({question, rating, additional_information}) => (
    {
        type: NEW_FEEDBACK_REQUEST,
        question,
        rating,
        additional_information,
    }
);