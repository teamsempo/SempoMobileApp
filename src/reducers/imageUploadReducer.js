export const RESET_IMAGE_UPLOAD = 'RESET_IMAGE_UPLOAD';
export const IMAGE_UPLOAD_REQUEST = 'IMAGE_UPLOAD_REQUEST';
export const IMAGE_UPLOAD_SUCCESS = 'IMAGE_UPLOAD_SUCCESS';
export const IMAGE_UPLOAD_FAILURE = 'IMAGE_UPLOAD_FAILURE';

export const initialImageUploadState = {
    isRequesting: false,
    success: false,
    imageData: null,
    error: null
};

export const imageUpload = (state = initialImageUploadState, action) => {
    switch (action.type) {
        case RESET_IMAGE_UPLOAD:
            return initialNewFeedbackState;
        case IMAGE_UPLOAD_REQUEST:
            return {...state, isRequesting: true, error: null, success: false};
        case IMAGE_UPLOAD_SUCCESS:
            return {...state, isRequesting: false, success: true};
        case IMAGE_UPLOAD_FAILURE:

            try {
                var error_message = action.error.response.data.message
            } catch (e) {
                error_message = 'unknown error'
            }

            return {...state, isRequesting: false, error: error_message};
        default:
            return state;
    }
};


// Actions

export const newImageUpload = (imageData, imageType, transferId = null) => (
    {
        type: IMAGE_UPLOAD_REQUEST,
        imageData: imageData,
        imageType: imageType,
        transferId: transferId
    }
);