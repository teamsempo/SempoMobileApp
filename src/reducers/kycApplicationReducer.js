import { DEEEEEEP } from "./utils";
import { combineReducers } from "redux";

export const UPDATE_KYC_DETAILS = 'UPDATE_KYC_DETAILS';
export const RESET_KYC_DETAILS = 'RESET_KYC_DETAILS';
export const UPDATE_KYC_APPLICATION_STATE = 'UPDATE_KYC_APPLICATION_STATE';

export const CREATE_KYC_APPLICATION_REQUEST = 'CREATE_KYC_APPLICATION_REQUEST';
export const CREATE_KYC_APPLICATION_SUCCESS = 'CREATE_KYC_APPLICATION_SUCCESS';
export const CREATE_KYC_APPLICATION_FAILURE = 'CREATE_KYC_APPLICATION_FAILURE';

export const LOAD_KYC_APPLICATION_REQUEST = 'LOAD_KYC_APPLICATION_REQUEST';
export const LOAD_KYC_APPLICATION_SUCCESS = 'LOAD_KYC_APPLICATION_SUCCESS';
export const LOAD_KYC_APPLICATION_FAILURE = 'LOAD_KYC_APPLICATION_FAILURE';

export const EDIT_KYC_APPLICATION_REQUEST = 'EDIT_KYC_APPLICATION_REQUEST';
export const EDIT_KYC_APPLICATION_SUCCESS = 'EDIT_KYC_APPLICATION_SUCCESS';
export const EDIT_KYC_APPLICATION_FAILURE = 'EDIT_KYC_APPLICATION_FAILURE';

export const UPLOAD_DOCUMENT_REQUEST = 'UPLOAD_DOCUMENT_REQUEST';
export const UPLOAD_DOCUMENT_SUCCESS = 'UPLOAD_DOCUMENT_SUCCESS';
export const UPLOAD_DOCUMENT_FAILURE = 'UPLOAD_DOCUMENT_FAILURE';

export const CREATE_BANK_ACCOUNT_REQUEST = 'CREATE_BANK_ACCOUNT_REQUEST';
export const CREATE_BANK_ACCOUNT_SUCCESS = 'CREATE_BANK_ACCOUNT_SUCCESS';
export const CREATE_BANK_ACCOUNT_FAILURE = 'CREATE_BANK_ACCOUNT_FAILURE';


// --- used for handling local state (device)
const initialKycDetails = {
    flowStep: 1,
    country: null,
    documentType: null,
    documentFrontUri: null, // image
    documentBackUri: null, // image
    selfieUri: null, // image

    // OLD flow.
    userDetails: null,
    addressDetails: null,
    paymentDetails: null,
};

const kycDetails = (state = initialKycDetails, action) => {
    switch (action.type) {
        case UPDATE_KYC_DETAILS:
            return DEEEEEEP(state, action.payload);
        case RESET_KYC_DETAILS:
            return initialKycDetails;
        default:
            return state;
    }
};

// --- used for handling server side state
const kycApplicationState = (state = {}, action) => {
    switch (action.type) {
        case UPDATE_KYC_APPLICATION_STATE:
            return DEEEEEEP(state, action.kyc_application);
        default:
            return state;
    }
};

const initialCreateStatusState = {
    isRequesting: false,
    error: null,
    success: false,
};

export const createStatus = (state = initialCreateStatusState, action) => {
    switch (action.type) {
        case CREATE_KYC_APPLICATION_REQUEST:
            return {...state, isRequesting: true, error: null, success: false};
        case CREATE_KYC_APPLICATION_SUCCESS:
            return {...state, isRequesting: false, success: true};
        case CREATE_KYC_APPLICATION_FAILURE:
            return {...state, isRequesting: false, error: action.error};
        default:
            return state;
    }
};

const initialLoadStatusState = {
    isRequesting: false,
    error: null,
    success: false
};

export const loadStatus = (state = initialLoadStatusState, action) => {
    switch (action.type) {
        case LOAD_KYC_APPLICATION_REQUEST:
            return {...state, isRequesting: true};

        case LOAD_KYC_APPLICATION_SUCCESS:
            return {...state, isRequesting: false, success: true};

        case LOAD_KYC_APPLICATION_FAILURE:
            return {...state, isRequesting: false, error: action.error};

        default:
            return state;
    }
};

export const initialEditStatusState = {
    isRequesting: false,
    error: null,
    success: false,
};

export const editStatus = (state = initialEditStatusState, action) => {
    switch (action.type) {
        case EDIT_KYC_APPLICATION_REQUEST:
            return {...state, isRequesting: true, error: null, success: false};
        case EDIT_KYC_APPLICATION_SUCCESS:
            return {...state, isRequesting: false, success: true};
        case EDIT_KYC_APPLICATION_FAILURE:
            return {...state, isRequesting: false, error: action.error};
        default:
            return state;
    }
};

export const initialUploadDocumentStatus = {
    isUploading: false,
    error: null,
    success: false
};

export const uploadDocumentStatus = (state = initialUploadDocumentStatus, action) => {
    switch (action.type) {
        case UPLOAD_DOCUMENT_REQUEST:
            return {...state, isUploading: true};

        case UPLOAD_DOCUMENT_SUCCESS:
            return {...state, isUploading: false, success: true};

        case UPLOAD_DOCUMENT_FAILURE:
            return {...state, isUploading: false, error: action.error};

        default:
            return state;
    }
};


export const initialCreateBankAccountStatus = {
    isRequesting: false,
    error: null,
    success: false,
};

export const createBankAccountStatus = (state = initialCreateBankAccountStatus, action) => {
    switch (action.type) {
        case CREATE_BANK_ACCOUNT_REQUEST:
            return {...state, isRequesting: true, error: null, success: false};
        case CREATE_BANK_ACCOUNT_SUCCESS:
            return {...state, isRequesting: false, success: true};
        case CREATE_BANK_ACCOUNT_FAILURE:
            return {...state, isRequesting: false, error: action.error};
        default:
            return state;
    }
};


export const kycApplication = combineReducers({
    kycDetails,
    kycApplicationState,
    uploadDocumentStatus,
    createBankAccountStatus,
    createStatus,
    loadStatus,
    editStatus,
});


// Actions
export const resetKYCDetails = (payload) => (
    {
        type: RESET_KYC_DETAILS,
        payload
    }
);

export const updateKYCDetails = (payload) => (
    {
        type: UPDATE_KYC_DETAILS,
        payload,
    }
);

export const loadKYCApplicationRequest = (payload) => (
    {
        type: LOAD_KYC_APPLICATION_REQUEST,
        payload
    }
);

export const createKYCApplication = (payload) => (
    {
        type: CREATE_KYC_APPLICATION_REQUEST,
        payload
    }
);

export const editKYCApplicationRequest = (payload) => (
    {
        type: EDIT_KYC_APPLICATION_REQUEST,
        payload
    }
);

export const uploadDocument = (payload) => (
    {
        type: UPLOAD_DOCUMENT_REQUEST,
        payload
    }
);

export const createBankAccount = (payload) => (
    {
        type: CREATE_BANK_ACCOUNT_REQUEST,
        payload
    }
);