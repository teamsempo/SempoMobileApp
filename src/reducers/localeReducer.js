export const LOCALE_REFRESH = 'LOCALE_REFRESH';
export const LOCALE_REQUEST = 'LOCALE_REQUEST';
export const LOCALE_SUCCESS = 'LOCALE_SUCCESS';
export const LOCALE_FAILURE = 'LOCALE_FAILURE';

export const initialLocaleState = {
    isRetrievingLocale: false,
    locale: null,
    isRTL: null,
};

export const locale = (state = initialLocaleState, action) => {
    switch (action.type) {
        case LOCALE_REFRESH:
            return {...state, isRetrievingLocale: true};
        case LOCALE_REQUEST:
            return {...state, isRetrievingLocale: true};
        case LOCALE_SUCCESS:
            return {...state, isRetrievingLocale: false, locale: action.locale, isRTL: action.isRTL};
        case LOCALE_FAILURE:
            return {...state, isRetrievingLocale: false, locale: null, isRTL: null, error: true};
        default:
            return state;
    }
};


// ACTIONS

export const localeRequest = (locale) => (
    {
        type: LOCALE_REQUEST,
        locale
    }
);