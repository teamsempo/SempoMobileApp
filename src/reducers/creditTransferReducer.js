import { combineReducers } from "redux";
import { DEEEEEEP } from "./utils";
import { IMAGE_UPLOAD_SUCCESS } from "./imageUploadReducer";

export const UPDATE_CREDIT_TRANSFER_LIST = "UPDATE_CREDIT_TRANSFER_LIST";

export const LOAD_CREDIT_TRANSFERS_REQUEST = 'LOAD_CREDIT_TRANSFERS_REQUEST';
export const LOAD_CREDIT_TRANSFERS_SUCCESS = 'LOAD_CREDIT_TRANSFERS_SUCCESS';
export const LOAD_CREDIT_TRANSFERS_FAILURE = 'LOAD_CREDIT_TRANSFERS_FAILURE';

export const UPDATE_TRANSFER_DATA = "UPDATE_TRANSFER_DATA";
export const RESET_TRANSFER_DATA = "RESET_TRANSFER_DATA";

export const CREATE_TRANSFER_REQUEST = 'CREATE_TRANSFER_REQUEST';
export const CREATE_TRANSFER_SUCCESS = 'CREATE_TRANSFER_SUCCESS';
export const CREATE_TRANSFER_FAILURE = 'CREATE_TRANSFER_FAILURE';

export const RESET_NEW_TRANSFER = 'RESET_NEW_TRANSFER';

export const RECORD_NFC_CARD_ERROR = 'RECORD_NFC_CARD_ERROR';
export const REMOVE_NFC_CARD_ERROR = 'REMOVE_NFC_CARD_ERROR';

export const byId = (state = {}, action) => {
    switch (action.type) {
        case UPDATE_CREDIT_TRANSFER_LIST:
            return DEEEEEEP(state, action.credit_transfers);

        case IMAGE_UPLOAD_SUCCESS:
            if (action.upload_result.data.uploaded_image.credit_transfer_id) {
                let newState = { ...state };
                newState[action.upload_result.data.uploaded_image.credit_transfer_id].attached_images.push(
                    action.upload_result.data.uploaded_image
                );
                return newState
            } else {
                return state
            }

        default:
            return state;
    }
};

export const initialLoadStatusState = {
    isRequesting: false,
    error: null,
    success: false,
    numberAdded: 1
};

export const loadStatus = (state = initialLoadStatusState, action) => {
    switch (action.type) {
        case LOAD_CREDIT_TRANSFERS_REQUEST:
            return { ...state, isRequesting: true };

        case LOAD_CREDIT_TRANSFERS_SUCCESS:
            return { ...state, isRequesting: false, success: true };

        case LOAD_CREDIT_TRANSFERS_FAILURE:
            return { ...state, isRequesting: false, error: action.error };
        case UPDATE_CREDIT_TRANSFER_LIST:
            return { ...state, numberAdded: Object.keys(action.credit_transfers).length };
        default:
            return state;
    }
};


export const initialTransferDataState = {
    user_id: null,
    transfer_amount: 0,
    transfer_use: null,
    transfer_random_key: null,
    is_sending: false,
    transfer_account_name: null,
    my_matching_transfer_accounts: null,
    my_transfer_account_id: null,
    token_symbol: null,

    // HANDLES PAYMENT FLOW
    counterparty_user_id: null,
    public_identifier: null,  // don't default reset as need for ETH transfers
    temp_transfer_mode: null,  // used for ETH transfers
    default_transfer_mode: null,

    // Catches NFC errors for retry capacity:
    errored_NFC_id: null
};

export const transferData = (state = initialTransferDataState, action) => {
    switch (action.type) {
        case UPDATE_TRANSFER_DATA:
            return DEEEEEEP(state, action.payload);
        case RESET_TRANSFER_DATA:  // DO NOT DEFAULT RESET default_transfer_mode, temp_transfer_mode or public_identifier
            return {
                ...state,
                user_id: null,
                transfer_amount: 0,
                transfer_use: null,
                transfer_random_key: null,
                transfer_account_name: null,
                counterparty_user_id: null,
                my_matching_transfer_accounts: null,
                errored_NFC_id: null
            };
        default:
            return state;
    }
};


export const initialCreateStatusState = {
    isRequesting: false,
    error: null,
    success: false,
    feedback: false,
};

export const createStatus = (state = initialCreateStatusState, action) => {
    switch (action.type) {
        case RESET_NEW_TRANSFER:
            return initialCreateStatusState;

        case CREATE_TRANSFER_REQUEST:
            return { ...state, isRequesting: true, error: null, success: false };

        case CREATE_TRANSFER_SUCCESS:
            if (action.result) {
                var feedback = action.result.feedback === true;
            } else {
                feedback = false
            }
            return { ...state, isRequesting: false, success: true, feedback: feedback };

        case CREATE_TRANSFER_FAILURE:
            return { ...state, isRequesting: false, error: action.error };

        default:
            return state;
    }
};

export const NFCError = (state = {}, action) => {
    switch (action.type) {
        case RECORD_NFC_CARD_ERROR:
            return {
                nfcSerialNumber: action.NFCSerialNumber,
                session: action.session,
                amountDeducted: action.amountDeducted,
                transferData: action.transferData
            };

        case REMOVE_NFC_CARD_ERROR:
            return {};

        default:
            return state;
    }
};

export const creditTransfers = combineReducers({
    byId,
    transferData,
    createStatus,
    loadStatus,
    NFCError
});

// ACTIONS

export const loadCreditTransfers = (per_page, page) => (
    {
        type: LOAD_CREDIT_TRANSFERS_REQUEST,
        payload: {
            query: {
                per_page,
                page
            }
        }
    }
);

export const resetTransferData = () => ({ type: RESET_TRANSFER_DATA });

export const updateTransferData = (payload) => (
    {
        type: UPDATE_TRANSFER_DATA,
        payload,
    }
);

export const createTransferRequest = (payload) => (
    {
        type: CREATE_TRANSFER_REQUEST,
        payload,
    }
);

export const removeNFCCardError = () => (
    {
        type: REMOVE_NFC_CARD_ERROR
    }
)
