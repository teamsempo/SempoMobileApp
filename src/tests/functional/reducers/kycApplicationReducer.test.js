import { createStatus, initialCreateStatusState } from '../../../reducers/kycApplicationReducer';
import { CREATE_KYC_APPLICATION_REQUEST, CREATE_KYC_APPLICATION_SUCCESS, CREATE_KYC_APPLICATION_FAILURE } from '../../../reducers/kycApplicationReducer';

describe('createStatus', () => {
    it('Returns correct state when createStatus is provided with CREATE_KYC_APPLICATION_REQUEST', () => {
        expect(createStatus(initialCreateStatusState, { type: CREATE_KYC_APPLICATION_REQUEST })).toMatchSnapshot();
    });

    it('Returns correct state when createStatus is provided with CREATE_KYC_APPLICATION_SUCCESS', () => {
        expect(createStatus(initialCreateStatusState, { type: CREATE_KYC_APPLICATION_SUCCESS })).toMatchSnapshot();
    });

    it('Returns correct state when createStatus is provided with CREATE_KYC_APPLICATION_FAILURE', () => {
        expect(createStatus(initialCreateStatusState, { type: CREATE_KYC_APPLICATION_FAILURE })).toMatchSnapshot();
    });
});

import { loadStatus, initialLoadStatusState } from '../../../reducers/kycApplicationReducer';
import { LOAD_KYC_APPLICATION_REQUEST, LOAD_KYC_APPLICATION_SUCCESS, LOAD_KYC_APPLICATION_FAILURE } from '../../../reducers/kycApplicationReducer';
describe('loadStatus', () => {
    it('Returns correct state when loadStatus is provided with LOAD_KYC_APPLICATION_REQUEST', () => {
        expect(loadStatus(initialLoadStatusState, { type: LOAD_KYC_APPLICATION_REQUEST })).toMatchSnapshot();
    });

    it('Returns correct state when loadStatus is provided with LOAD_KYC_APPLICATION_SUCCESS', () => {
        expect(loadStatus(initialLoadStatusState, { type: LOAD_KYC_APPLICATION_SUCCESS })).toMatchSnapshot();
    });

    it('Returns correct state when loadStatus is provided with LOAD_KYC_APPLICATION_FAILURE', () => {
        expect(loadStatus(initialLoadStatusState, { type: LOAD_KYC_APPLICATION_FAILURE })).toMatchSnapshot();
    });
});

import { editStatus, initialEditStatusState } from '../../../reducers/kycApplicationReducer';
import { EDIT_KYC_APPLICATION_REQUEST, EDIT_KYC_APPLICATION_SUCCESS, EDIT_KYC_APPLICATION_FAILURE } from '../../../reducers/kycApplicationReducer';
describe('editStatus', () => {
    it('Returns correct state when editStatus is provided with EDIT_KYC_APPLICATION_REQUEST', () => {
        expect(editStatus(initialEditStatusState, { type: EDIT_KYC_APPLICATION_REQUEST })).toMatchSnapshot();
    });

    it('Returns correct state when editStatus is provided with EDIT_KYC_APPLICATION_SUCCESS', () => {
        expect(editStatus(initialEditStatusState, { type: EDIT_KYC_APPLICATION_SUCCESS })).toMatchSnapshot();
    });

    it('Returns correct state when editStatus is provided with EDIT_KYC_APPLICATION_FAILURE', () => {
        expect(editStatus(initialEditStatusState, { type: EDIT_KYC_APPLICATION_FAILURE })).toMatchSnapshot();
    });
});

import { uploadDocumentStatus, initialUploadDocumentStatus } from '../../../reducers/kycApplicationReducer';
import { UPLOAD_DOCUMENT_REQUEST, UPLOAD_DOCUMENT_SUCCESS, UPLOAD_DOCUMENT_FAILURE } from '../../../reducers/kycApplicationReducer';
describe('uploadDocumentStatus', () => {
    it('Returns correct state when uploadDocumentStatus is provided with UPLOAD_DOCUMENT_REQUEST', () => {
        expect(uploadDocumentStatus(initialUploadDocumentStatus, { type: UPLOAD_DOCUMENT_REQUEST })).toMatchSnapshot();
    });

    it('Returns correct state when uploadDocumentStatus is provided with UPLOAD_DOCUMENT_SUCCESS', () => {
        expect(uploadDocumentStatus(initialUploadDocumentStatus, { type: UPLOAD_DOCUMENT_SUCCESS })).toMatchSnapshot();
    });

    it('Returns correct state when uploadDocumentStatus is provided with UPLOAD_DOCUMENT_FAILURE', () => {
        expect(uploadDocumentStatus(initialUploadDocumentStatus, { type: UPLOAD_DOCUMENT_FAILURE })).toMatchSnapshot();
    });
});

import { createBankAccountStatus, initialCreateBankAccountStatus } from '../../../reducers/kycApplicationReducer';
import { CREATE_BANK_ACCOUNT_REQUEST, CREATE_BANK_ACCOUNT_SUCCESS, CREATE_BANK_ACCOUNT_FAILURE } from '../../../reducers/kycApplicationReducer';
describe('createBankAccountStatus', () => {
    it('Returns correct state when createBankAccountStatus is provided with CREATE_BANK_ACCOUNT_REQUEST', () => {
        expect(createBankAccountStatus(initialCreateBankAccountStatus, { type: CREATE_BANK_ACCOUNT_REQUEST })).toMatchSnapshot();
    });

    it('Returns correct state when createBankAccountStatus is provided with CREATE_BANK_ACCOUNT_SUCCESS', () => {
        expect(createBankAccountStatus(initialCreateBankAccountStatus, { type: CREATE_BANK_ACCOUNT_SUCCESS })).toMatchSnapshot();
    });

    it('Returns correct state when createBankAccountStatus is provided with CREATE_BANK_ACCOUNT_FAILURE', () => {
        expect(createBankAccountStatus(initialCreateBankAccountStatus, { type: CREATE_BANK_ACCOUNT_FAILURE })).toMatchSnapshot();
    });
});
