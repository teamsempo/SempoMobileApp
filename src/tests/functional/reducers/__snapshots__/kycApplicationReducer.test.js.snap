// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBankAccountStatus Returns correct state when createBankAccountStatus is provided with CREATE_BANK_ACCOUNT_FAILURE 1`] = `
Object {
  "error": undefined,
  "isRequesting": false,
  "success": false,
}
`;

exports[`createBankAccountStatus Returns correct state when createBankAccountStatus is provided with CREATE_BANK_ACCOUNT_REQUEST 1`] = `
Object {
  "error": null,
  "isRequesting": true,
  "success": false,
}
`;

exports[`createBankAccountStatus Returns correct state when createBankAccountStatus is provided with CREATE_BANK_ACCOUNT_SUCCESS 1`] = `
Object {
  "error": null,
  "isRequesting": false,
  "success": true,
}
`;

exports[`createStatus Returns correct state when createStatus is provided with CREATE_KYC_APPLICATION_FAILURE 1`] = `
Object {
  "error": undefined,
  "isRequesting": false,
  "success": false,
}
`;

exports[`createStatus Returns correct state when createStatus is provided with CREATE_KYC_APPLICATION_REQUEST 1`] = `
Object {
  "error": null,
  "isRequesting": true,
  "success": false,
}
`;

exports[`createStatus Returns correct state when createStatus is provided with CREATE_KYC_APPLICATION_SUCCESS 1`] = `
Object {
  "error": null,
  "isRequesting": false,
  "success": true,
}
`;

exports[`editStatus Returns correct state when editStatus is provided with EDIT_KYC_APPLICATION_FAILURE 1`] = `
Object {
  "error": undefined,
  "isRequesting": false,
  "success": false,
}
`;

exports[`editStatus Returns correct state when editStatus is provided with EDIT_KYC_APPLICATION_REQUEST 1`] = `
Object {
  "error": null,
  "isRequesting": true,
  "success": false,
}
`;

exports[`editStatus Returns correct state when editStatus is provided with EDIT_KYC_APPLICATION_SUCCESS 1`] = `
Object {
  "error": null,
  "isRequesting": false,
  "success": true,
}
`;

exports[`loadStatus Returns correct state when loadStatus is provided with LOAD_KYC_APPLICATION_FAILURE 1`] = `
Object {
  "error": undefined,
  "isRequesting": false,
  "success": false,
}
`;

exports[`loadStatus Returns correct state when loadStatus is provided with LOAD_KYC_APPLICATION_REQUEST 1`] = `
Object {
  "error": null,
  "isRequesting": true,
  "success": false,
}
`;

exports[`loadStatus Returns correct state when loadStatus is provided with LOAD_KYC_APPLICATION_SUCCESS 1`] = `
Object {
  "error": null,
  "isRequesting": false,
  "success": true,
}
`;

exports[`uploadDocumentStatus Returns correct state when uploadDocumentStatus is provided with UPLOAD_DOCUMENT_FAILURE 1`] = `
Object {
  "error": undefined,
  "isUploading": false,
  "success": false,
}
`;

exports[`uploadDocumentStatus Returns correct state when uploadDocumentStatus is provided with UPLOAD_DOCUMENT_REQUEST 1`] = `
Object {
  "error": null,
  "isUploading": true,
  "success": false,
}
`;

exports[`uploadDocumentStatus Returns correct state when uploadDocumentStatus is provided with UPLOAD_DOCUMENT_SUCCESS 1`] = `
Object {
  "error": null,
  "isUploading": false,
  "success": true,
}
`;
